mesonbuild/cmake/toolchain.py

```diff
@@ -198,7 +199,7 @@ def is_cmdline_option(compiler: 'Compiler', arg: str) -> bool:
         if compiler.get_argument_syntax() == 'msvc':
             return arg.startswith('/')
         else:
-            if compiler.exelist[0] == 'zig' and arg in {'ar', 'cc', 'c++', 'dlltool', 'lib', 'ranlib', 'objcopy', 'rc'}:
+            if os.path.basename(compiler.get_exe()) == 'zig' and arg in {'ar', 'cc', 'c++', 'dlltool', 'lib', 'ranlib', 'objcopy', 'rc'}:
```

Comment:
This change appears to be unrelated to creating `get_exe()` but is saying that the original zig handling from eda59e141d30204908ef1f56b2637164310c9ccc was flawed: it assumes that people only use zig via `CC="zig cc"` and never, say, `CC="/usr/bin/zig cc"`.

(I have no idea what people actually do here, but it's definitely broken at the theoretical model, yeah.)

Reply:
Yes, `exelist[0]` becomes `get_exe()` but the result is clearly broken.